### PR TITLE
Add IPublicClientApplication interface

### LINKED ISSUE #1743: Add interface for PublicClientApplication
Please follow the issue template below. 

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Upcoming Library
- [ ] `@azure/msal-node@1.x`

## Source
- [x] Internal (MSAL.js team)
- [ ] Customer request

## Description

We should create and export an interface for PublicClientApplication, e.g. `IPublicClientApplication`. This should document what are public API is for PublicClientApplication, the main class for the library. This will also be useful for use cases such as MSAL React.

## Priority

Medium

## Acceptance Criteria

1. An interface for PublicClientApplication is available as an export.

## Prior art

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/5eb0bc9fc7c962bc293aac62e0e51f78ae939c22/samples/msal-react-sample/src/msal-react/index.tsx#L21-L31

Related:

#1427

### PR DESCRIPTION
**Pull request by jasonnutter**
Adds an `IPublicClientApplication` interface, properly documenting what the supported public API for `PublicClientApplication` will be.

### CODE REVIEW

File: lib/msal-browser/src/app/PublicClientApplication.ts [hunk 1]
```diff
@@ -452,7 +452,7 @@ export class PublicClientApplication {
      *
      * @returns {string} post logout redirect URL
      */
-    public getPostLogoutRedirectUri(requestPostLogoutRedirectUri?: string): string {
+    private getPostLogoutRedirectUri(requestPostLogoutRedirectUri?: string): string {
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/PublicClientApplication.ts:455]: These do not need to be public, so making them private.

File: lib/msal-browser/src/app/IPublicClientApplication.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { AuthenticationResult, AuthorizationUrlRequest } from "@azure/msal-common";
+import { SilentFlowRequest, EndSessionRequest, AccountInfo } from "../";
+
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+export interface IPublicClientApplication {
```
Comment (DarylThayil) [inline lib/msal-browser/src/app/IPublicClientApplication.ts:8]: Shouldn't the return types be here?

Comment (jasonnutter) [reply lib/msal-browser/src/app/IPublicClientApplication.ts:8]: They are: `functionName(params): ReturnType;`

File: lib/msal-browser/src/app/PublicClientApplication.ts [hunk 3]
```diff
@@ -35,16 +35,16 @@ import { RedirectHandler } from "../interaction_handler/RedirectHandler";
 import { PopupHandler } from "../interaction_handler/PopupHandler";
 import { SilentHandler } from "../interaction_handler/SilentHandler";
 import { BrowserAuthError } from "../error/BrowserAuthError";
-import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/PublicClientApplication.ts:38]: Unused.

File: lib/msal-browser/src/app/IPublicClientApplication.ts (hunk 2 above)
Comment (DarylThayil) [reply lib/msal-browser/src/app/IPublicClientApplication.ts:8]: How did I miss that .... haha

**Review by DarylThayil — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
lgtm

**Review by pkanher617 — APPROVED** [review]

**Pull request merged**